servlet/src/main/java/io/undertow/servlet/spec/PushBuilderImpl.java

```diff
@@ -112,15 +112,14 @@ public PushBuilderImpl(HttpServletRequestImpl servletRequest) {
             if(cookie.getValue().getMaxAge() != null && cookie.getValue().getMaxAge() <= 0) {
                 //remove cookie
                 List<String> existing = this.headers.getAll(HttpHeaderNames.COOKIE);
-                if(existing != null) {
-                    Iterator<String> it = existing.iterator();
-                    while (it.hasNext()) {
-                        String val = it.next();
-                        if(val.startsWith(cookie.getKey() + "=")) {
-                            it.remove();
-                        }
+                List<String> newValues = new ArrayList<>(existing.size());
+                for (String value : existing) {
+                    if(value.startsWith(cookie.getKey() + "=")) {
+                        continue;
                     }
+                    newValues.add(value);
                 }
+                this.headers.set(HttpHeaderNames.COOKIE, newValues);
```

Comment:
Does it erase the older value or suffix it (headers are multimap), I believe `set` erase, so it would work, but it needs to be checked.

Reply:
```
Sets a header with the specified name and values. If there is an existing header with the same name, it is removed
```